pyiron_atomistics/atomistics/structure/analyse.py

```diff
@@ -689,3 +689,63 @@ def get_delaunay_neighbors(self, width_buffer: float = 10.) -> np.ndarray:
             pairs (ndarray): Delaunay neighbor indices
         """
         return self._get_neighbors(Delaunay, "simplices", width_buffer=width_buffer)
+
+    def cluster_positions(self, positions=None, eps=1, buffer_width=None, return_labels=False):
+        """
+        Cluster positions according to the distances. Clustering algorithm uses DBSCAN:
+
+        https://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html
+
+        Example I:
+
+        ```
+        analyse = Analyze(some_pyiron_structure)
+        positions = analyse.cluster_points(eps=2)
+        ```
+
+        This example should return the atom positions, if no two atoms lie within a distance of 2.
+        If there are at least two atoms which lie within a distance of 2, their entries will be
+        replaced by their mean position.
+
+        Example II:
+
+        ```
+        analyse = Analyze(some_pyiron_structure)
+        print(analyse.cluster_positions([3*[0.], 3*[1.]], eps=3))
+        ```
+
+        This returns `[0.5, 0.5, 0.5]` (if the cell is large enough)
+
+        Args:
+            positions (numpy.ndarray): Positions to consider. Default: atom positions
+            eps (float): The maximum distance between two samples for one to be considered as in
+                the neighborhood of the other.
+            buffer_width (float): Buffer width to consider across the periodic boundary
+                conditions. If too small, it is possible that atoms that are meant to belong
+                together across PBC are missed. Default: Same as eps
+            return_labels (bool): Whether to return the labels given according to the grouping
+                together with the mean positions
+
+        Returns:
+            positions (numpy.ndarray): Mean positions
+            label (numpy.ndarray): Labels of the positions (returned when `return_labels = True`)
+        """
+        if positions is None:
+            positions = self._structure.positions
```

Comment:
```suggestion
        positions = self._structure.positions if positions is None else np.array(positions)
```

I guess the structure.positions are already safely an array so we can one-line this no problem